src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -288,7 +341,107 @@ def get_libsvm_dmatrix(files_path):
     return dmatrix
 
 
-def get_dmatrix(data_path, content_type, csv_weights=0):
+def get_parquet_dmatrix(files_path):
+    try:
+        table = pq.read_table(files_path)
+
+        data = table.to_pandas()
+        if type(data) is pd.DataFrame:
+            # pyarrow.Table.to_pandas may produce NumPy array or pandas DataFrame
+            data = data.to_numpy()
+
+        dmatrix = xgb.DMatrix(data[:, 1:], label=data[:, 0])
+        return dmatrix
+
+    except Exception as e:
+        raise exc.UserError("Failed to load parquet data with exception:\n{}".format(e))
+
+
+def get_parquet_dmatrix_pipe_mode(pipe_path):
+    try:
+        f = mlio.SageMakerPipe(pipe_path)
+        examples = []
+
+        with f.open_read() as strm:
+            reader = mlio.ParquetRecordReader(strm)
+
+            for record in reader:
+                table = pq.read_table(as_arrow_file(record))
+                array = table.to_pandas()
+                if type(array) is pd.DataFrame:
+                    array = array.to_numpy()
+                examples.append(array)
+
+        if examples:
+            data = np.vstack(examples)
+            del examples
+
+            dmatrix = xgb.DMatrix(data[:, 1:], label=data[:, 0])
+            return dmatrix
+        else:
+            return None
+
+    except Exception as e:
+        raise exc.UserError("Failed to load parquet data with exception:\n{}".format(e))
+
+
+def get_recordio_protobuf_dmatrix(files_path):
```

Comment:
I assume that you are going to merge `get_recordio_protobuf_dmatrix` and `get_recordio_protobuf_dmatrix_pipe_mode` at some point.

Reply:
I was going to leave them separate. Do you think it makes more sense to have the separate file mode and pipe mode functions all merged (i.e. only have one method per content type), or are you only pointing out recordio-protobuf in this case because the code is nearly identical?

Reply:
It was mainly to follow the convention of the other functions, which are different enough to warrant two separate functions. In this case I'll combine it into a single function.

Reply:
What I could also do is have a single interface for both file mode and pipe mode functions, which actually sounds like a more appropriate design here.